### ISSUE TITLE
Error in loading training set of VOCDataset of 2007 

### ISSUE THREAD

**Issue by AmrKhalifa** [pytorch/vision#3585]
## 🐛 Bug

Downloading training set of VOCDetection 2007 version using torchvision.datasets downloads the test set. 

## To Reproduce

1 - Download the training set of VOCDetection 2007 version. 

torchvision.datasets.VOCDetection(image_root, year='2007', image_set="train",
                                    download=True) 

this is what's gets downloaded:
Downloading http://host.robots.ox.ac.uk/pascal/VOC/voc2007/VOCtest_06-Nov-2007.tar

while it should download the training part 
this is due to wrongly handling of assigning key value in _VOCBase class 
https://github.com/pytorch/vision/blob/master/torchvision/datasets/voc.py 

## Environment

 PyTorch version: 1.6.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 11.2 (x86_64)
GCC version: Could not collect
Clang version: 12.0.0 (clang-1200.0.32.27)
CMake version: Could not collect

Python version: 3.8 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.18.5
[pip3] numpydoc==1.1.0
[pip3] torch==1.6.0
[pip3] torchvision==0.7.0
[conda] blas                      1.0                         mkl
[conda] mkl                       2019.4                      233
[conda] mkl-service               2.3.0            py38hfbe908c_0
[conda] mkl_fft                   1.1.0            py38hc64f4ea_0
[conda] mkl_random                1.1.1            py38h959d312_0
[conda] numpy                     1.18.5           py38h1da2735_0
[conda] numpy-base                1.18.5           py38h3304bdc_0
[conda] numpydoc                  1.1.0                      py_0
[conda] torch                     1.6.0                    pypi_0    pypi
[conda] torchvision               0.7.0                    pypi_0    pypi

**Comment by pmeier** [comment]
Duplicate of #3571.
